src/App.tsx

```diff
@@ -1,24 +1,141 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useState, useMemo,
+} from 'react';
+import cn from 'classnames';
+import { Todo, EnumTodoFilter } from './types/types';
+import { getTodos, deleteTodo, USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoForm } from './components/TodoForm';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(EnumTodoFilter.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const isExistCompleted = completedTodos.length > 0;
+
+  const getFilteredTodos = useCallback((filter: EnumTodoFilter) => {
+    switch (filter) {
+      case EnumTodoFilter.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      case EnumTodoFilter.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [filterBy, todos]);
+
+  const createTodo = (newTodo: Todo) => {
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const deleteErrorMessage = useCallback(() => {
+    setErrorMessage('');
+  }, []);
+
+  const loadData = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorMessage('Failed to load data');
+    }
+  }, []);
+
+  const handleTodoDelete = useCallback(async (todoToDelete: Todo) => {
+    try {
+      setErrorMessage('');
+      setTempTodo(todoToDelete);
+      await deleteTodo(todoToDelete.id);
+      loadData();
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setTempTodo(null);
+    }
+  }, []);
+
+  const handleClearCompleted = useCallback(() => {
+    for (let i = 0; i < completedTodos.length; i += 1) {
+      handleTodoDelete(completedTodos[i]);
+    }
```

Comment:
let's use `Promise.all()`, ensure `async`/`await` (since the function called is async itself)
also such a vanile for — a kind of not advanced